.github/workflows/lint.yml

```diff
@@ -0,0 +1,46 @@
+# Run various link checks, like flake8 and black, to make sure
+# our code remains in good shape, avoids common bugs, and follows
+# common coding  conventions.
+name: lint
+
+on:
+  push:
+    branches-ignore:
+      - main
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  linting:
+    runs-on: ubuntu-latest
+    steps:
+      #----------------------------------------------
+      #       check-out repo and set-up python
+      #----------------------------------------------
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
```

Comment:
@vengroff should this be 3.10? I just copied from dohyo.

Reply:
It's the version for running the lint tools, not our actual code, but probably should be 3.10 to avoid confusion like this.